packages/integration-tests/src/test-e2e/iam_access_drift.test.ts

```diff
@@ -0,0 +1,348 @@
+    const policies: Array<RoleInlinePolicy> = [];
+    const roleNames = await deployedResourcesFinder.findByBackendIdentifier(
+      branchBackendIdentifier,
+      'AWS::IAM::Role'
+    );
+    for (const roleName of roleNames) {
+      let nextToken: string | undefined;
+      do {
+        const listRolePoliciesResponse: ListRolePoliciesCommandOutput =
+          await iamClient.send(
+            new ListRolePoliciesCommand({
+              RoleName: roleName,
+              Marker: nextToken,
+            })
+          );
+        nextToken = listRolePoliciesResponse.Marker;
+        if (listRolePoliciesResponse.PolicyNames) {
+          for (const policyName of listRolePoliciesResponse.PolicyNames) {
+            const getRolePolicyResponse: GetRolePolicyCommandOutput =
+              await iamClient.send(
+                new GetRolePolicyCommand({
+                  RoleName: roleName,
+                  PolicyName: policyName,
+                })
+              );
+            if (getRolePolicyResponse.PolicyDocument) {
+              policies.push({
+                roleName,
+                policyName,
+                policyStatement: decodeURIComponent(
+                  getRolePolicyResponse.PolicyDocument
+                ),
+              });
+            }
+          }
+        }
+      } while (nextToken);
+    }
+    return policies;
+  };
+
+  const reinstallDependencies = async (): Promise<void> => {
+    await fsp.rm(path.join(tempDir, 'node_modules'), {
+      recursive: true,
+      force: true,
+    });
+    await fsp.unlink(path.join(tempDir, 'package-lock.json'));
+
+    await execa('npm', ['install'], {
+      cwd: tempDir,
+      stdio: 'inherit',
+    });
+  };
+
+  const comparePolicy = async (
+    baselinePolicy: string,
+    currentPolicy: string,
+    policyType: AccessCheckPolicyType
+  ): Promise<CheckNoNewAccessCommandOutput> => {
+    return await accessAnalyzerClient.send(
+      new CheckNoNewAccessCommand({
+        existingPolicyDocument: baselinePolicy,
+        newPolicyDocument: currentPolicy,
+        policyType: policyType,
+      })
+    );
+  };
+
+  void it('should not drift iam policies', async () => {
+    await baselineNpmProxyController.setUp();
+
+    await execa('npm', ['create', amplifyAtTag, '--yes'], {
+      cwd: tempDir,
+      stdio: 'inherit',
+    });
+
+    await deploy();
+
+    const baselineManagedIamPolicies = await listManagedIamPolicies();
+    const baselineRoleTrustPolicies = await listRoleTrustPolicies();
+    const baselineRoleInlinePolicies = await listRoleInlinePolicies();
+
+    await baselineNpmProxyController.tearDown();
+    await currentNpmProxyController.setUp();
+    await reinstallDependencies();
+
+    await deploy();
+
+    const currentManagedIamPolicies = await listManagedIamPolicies();
+    const currentRoleTrustPolicies = await listRoleTrustPolicies();
+    const currentRoleInlinePolicies = await listRoleInlinePolicies();
```

Comment:
We should also check the size of both policy arrays. E.g. if there is an extra `current` policy, it will not be caught by the comparisons below.